language-support/hs/bindings/src/DA/Ledger/Convert.hs

```diff
@@ -354,7 +355,7 @@ raiseValue = \case
         LL.ValueSumContractId c -> (return . VContract . ContractId) c
         LL.ValueSumList vs -> (fmap VList . raiseList raiseValue . LL.listElements) vs
         LL.ValueSumInt64 i -> (return . VInt . fromIntegral) i
-        LL.ValueSumDecimal t -> (return . VDecimal) t
+        LL.ValueSumDecimal t -> (return . VDecimal . read . Text.unpack) t
```

Comment:
What happens if the string has more than 10 digits after the period?

Reply:
I imagine it will fail.
What does the Ledger API guarantee?

Reply:
Failing would be good, swallowing digits or rounding would be slightly scary. The current Ledger API never gives you more than 10 digits but this might change in the future. Crashing loudly if it does would be nice.

Reply:
I did some experiments: -- `read "123.11223344556677"`
Turns out that it does silently swallow digits! --> `123.1122334455`
But it seems this is consistent with Haskell's overall approach to fixed width decimals..
For example, writing `VDecimal 123.11223344556677`, and so invoking the fractional instance, performs the exact same truncation.

Reply:
I don’t think being consistent with Haskell’s default approach is useful in any way here. Throwing an error is a much better option.

Reply:
I wasn't suggesting it was, I was just reporting what I discovered.

Reply:
I tried coded something which for any `a` with `Read a` and `Show a` which checks round-tripability: 
```
readCheckRoundTrip :: (Read a,Show a) => String -> a
readCheckRoundTrip s = if s /= s' then error msg else a
  where
      a = read s
      s' = show a
      msg = "read does not round-trip correctly: [" <> s <> "] --> [" <> s' <> "]"
```

but is not right when there are multiple `String` reps for the same `a`, i.e. trailing `0`s when printing a fixed width number.

So perhaps we should avoid coding what surely must be a solved problem...

Do you know if there is any standard support for `Fixed`  which allows checking if `read` is *safe*, i.e. does not loose precision?

Reply:
Sadly, I don’t know of any package that provides this :disappointed: